be/src/util/arrow/starrocks_column_to_arrow.cpp

```diff
@@ -24,6 +26,45 @@
 
 namespace starrocks {
 
+class ColumnContext;
+
+// Function to convert the column data in the range [start_idx, end_idx] to arrow
+typedef arrow::Status (*StarRocksToArrowConvertFunc)(const ColumnPtr& column, int start_idx, int end_idx,
+                                                     ColumnContext* column_context, arrow::ArrayBuilder* array_builder);
+
+// Resolve the convert function for the column according to types and nullable
+static inline StarRocksToArrowConvertFunc resolve_convert_func(LogicalType lt, ArrowTypeId at, bool is_nullable);
+
+// The context for a column
+struct ColumnContext {
+    ColumnContext(const TypeDescriptor& desc, const std::shared_ptr<arrow::DataType>& at,
+                  StarRocksToArrowConvertFunc func)
+            : type_desc(desc), arrow_type(at), convert_func(func) {}
+
+    const TypeDescriptor& type_desc;
+    const std::shared_ptr<arrow::DataType>& arrow_type;
+
+    // Function used to convert this column to arrow
+    StarRocksToArrowConvertFunc convert_func;
+
+    // Child column contexts which are only valid for Array/Struct/Map column.
+    // The vector is lazily initialized, and should check before using it.
+    // The size of the vector depends on the type of the column
+    //    Array:  one context for the element column of ArrayColumn
+    //    Struct: contexts for each field column of StructColumn
+    //    Map:    two contexts for key and value columns of MapColumn
+    bool child_context_initialized{false};
+    std::vector<ColumnContext> child_column_contexts;
+};
+
+static inline arrow::Status check_const_and_only_null(const ColumnPtr& column) {
+    if (!column->is_constant() && !column->only_null()) {
+        return arrow::Status::OK();
+    }
+    return column->is_constant() ? arrow::Status::Invalid(fmt::format("The column can not be constant"))
+                                 : arrow::Status::Invalid(fmt::format("The column can not only have null values"));
+}
```

Comment:
only null column is  a sepcial const col.

Reply:
Fixed